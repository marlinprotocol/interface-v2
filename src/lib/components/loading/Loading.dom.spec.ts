import { describe, it, expect } from 'vitest';
import { render } from '@testing-library/svelte';
import LoaderAnimatedPing from './LoadingAnimatedPing.svelte'; // Replace with the correct path to your component

describe('Loader Animated Ping', () => {
	it('renders loader with two child div elements', () => {
		const { container } = render(LoaderAnimatedPing);

		const loader = container.querySelector('.marlin-loader');
		expect(loader).toBeTruthy();
		expect(loader?.className).contain('marlin-loader');

		const rippleDivs = container.querySelectorAll('.lds-ripple div');
		// There should be two divs for the ripple effect.
		expect(rippleDivs.length).toBe(2);

		// Snapshot testing to ensure consistent render
		expect(container).toMatchSnapshot();
	});
});
