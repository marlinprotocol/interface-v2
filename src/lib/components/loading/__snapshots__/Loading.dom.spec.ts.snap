// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Loader Animated Ping > renders loader with two child div elements 1`] = `
<body>
  <div>
    <div
      class="marlin-loader s-pih5pjlcIrgJ"
    >
      <div
        class="lds-ripple s-pih5pjlcIrgJ"
      >
        <div
          class="s-pih5pjlcIrgJ"
        />
         
        <div
          class="s-pih5pjlcIrgJ"
        />
      </div>
    </div>
    <!--&lt;LoadingAnimatedPing&gt;-->
  </div>
</body>
`;
